tools/datagen/src/bin/datagen.rs

```diff
@@ -322,10 +322,8 @@ fn get_blob_exporter(matches: &ArgMatches) -> anyhow::Result<BlobExporter<'stati
 
     let output_path: Option<PathBuf> = if matches.is_present("OUTPUT_TESTDATA") {
         Some(icu_testdata::paths::data_root().join("testdata.postcard"))
-    } else if let Some(v) = matches.value_of_os("OUTPUT") {
-        Some(PathBuf::from(v))
     } else {
-        None
+        matches.value_of_os("OUTPUT").map(PathBuf::from)
     };
```

Comment:
Nit (non-actionable): I find the previous code more readable, but I don't care enough to override clippy.